src/imitation/rewards/discrim_net.py

```diff
@@ -75,6 +63,79 @@ def build_graph(self):
     FIXME(sam): remove train reward/test reward distinction.
     """
 
+  def _add_default_train_stats(self) -> None:
+    """Updates `self._train_stats` with several default summary Tensors.
+
+    Also creates a `tf.summary.histogram` for the `disc_logits`
+    argument.
+
+    Params:
+      disc_logits: A rank-1 Tensor containing binary logit for expert
+        (positive value) versus generator (negative values).
+    """
+    tf.summary.histogram("disc_logits", self._disc_logits)
+    bin_is_generated_pred = self._disc_logits > 0
```

Comment:
Is this the right way around for AIRL? With the current AIRL code, `_disc_logits = self.reward_net.reward_output_train, self.log_policy_act_prob_ph`, so a high discriminator value suggests that the sample is from an expert, rather than the imitator. In GAIL I believe it's reversed, so high discriminator output is an indicator of fakeness (in which case this code does make sense—see eqn. (16) & exposition after it in https://arxiv.org/pdf/1606.03476.pdf to figure out which way sign should go for GAIL). It is worth running both AIRL & GAIL using these stats to verify that they move in the right direction in both cases.

Reply:
Oops, ignore the above comment! I believe got the GAIL paper the wrong way around. Looks like the policy in GAIL is treating `\log D(s,a)` as a cost, so it tries to drive it _down_. I misread it originally as a reward function. Interpretation of `\log D(s,a)` is consistent with eqn (14) in the paper.

Reply:
`label_ph` always forces expert (0.0) and generated (1.0), and we see at least half-decent training in both algorithms.

In the definition of DiscrimNet.reward_train:

https://github.com/HumanCompatibleAI/imitation/blob/0331f0edb1f3e85f3d86bfd5ad89027656cef305/src/imitation/rewards/discrim_net.py#L121

So I don't think that AIRL/GAIL are flipped in our implementation. (Continuing to look at this more thoroughly now).

Reply:
Okay, thanks for checking that it works anyway. Sorry for the misleading comment earlier.